lightningd/htlc_set.c

```diff
@@ -181,8 +181,7 @@ void htlc_add_set(struct lightningd *ld,
 	}
 
 	if (amount_msat_eq(set->so_far, total_msat)) {
-		/* FIXME: hand to invoice_try_pay! */
-		tal_free(set);
+		invoice_try_pay(ld, set, details);
```

Comment:
Should this be `take(set)` for self-documentation?

Reply:
Hmm, no, it *has* to dispose of it.  take() is for things where it may or may not take ownership.

We've generally used naming to try to indicate which functions are a sink.  Should we rename this to 'invoice_resolve_htlc_in'?

Reply:
That sounds like a good idea. I keep getting confused by `take()` :-)

I'm tempted to propose an `OWNS` annotation and an `own()` nop-helper that just serves as annotation to say: I'll take ownership no matter what and you MUST take ownership. Purely documentation but we'd know which side of the call is wrong if we have a leak without having to dig through both the caller and callee to see what the semantics should have been. Just fantasizing here of course :smiley: